### PR TITLE
Limb damage rework

### LINKED ISSUE #38445: Surplus prosthetics are armored
1. Grab a wrench
2. Smack prosthetic limb repeatedly, take no damage

Anything with force 5 and under does no damage. I doubt this is intentional as the description specifically states it's supposed to be fragile.

### PR DESCRIPTION
**Pull request by XDTM**
:cl: XDTM
balance: Limbs no longer need to have full damage to be dismemberable, although more damage means higher dismemberment chance.
balance: Damaging limbs now only counts as 75% for the mob's total health. Bleeding still applies normally.
add: Limbs that reach max damage will now be disabled until they are healed to half health.
add: Disabled limbs work similarly to missing limbs, but they can still be used to wear items or handcuffs.
/:cl:

Fixes #38445

<s>CODE IS STILL UNTESTED - WILL TEST WHEN I'M ABLE TO</s>
TESTED, WORKIN' FINE

Always felt odd how reaching max damage on a limb had no consequence. Now it does.
Currently i only added effects for disabling arms and legs; "breaking" heads and torsos has no effect as of now.

### CODE REVIEW

**Comment by XDTM** [comment]
It's not like Deus Ex is our benchmark for mechanics, but anyway it's to avoid limb functionality flickering on and off from periodic damage ticks, like having bicaridine while spacewalking. Plus it makes disabling a limb last for at least a while, making it worthwhile.

**Comment by cacogen** [comment]
I wasn't implying that it was. It just doesn't make sense to me that a limb wouldn't start working again at the point before it stopped.

**Comment by Tlaltecuhtli** [comment]
so uhh did you test if eswords on leg are op now?

**Comment by XDTM** [comment]
The answer is in all caps in the PR body

**Comment by Tlaltecuhtli** [comment]
better test it

**Comment by Ratvar** [comment]
So, species with no dismemberment will be technically dismembered if toolboxed anyway? Seems like a nerf to abductors, golems and everything else blahblah

**Comment by XDTM** [comment]
Limbs have 50 health right now, so they can be disabled without crit. Taking any more damage on that limb has no effect, like now.
I'll add the immunity check for disabling as well.

**Comment by Ratvar** [comment]
Toolbox someone's legs and then heal them to 20 health for permastun?

**Comment by Ratvar** [comment]
PS: This will buff eswords/armblades and all other stuff that was supposed to cut limbs off because "dismembering" is now guaranteed.
Dismembering weapons are now worse then non-dismembering because people will stop bleeding/regain health when limb is cut off.

**Comment by XDTM** [comment]
You seem to think that reattaching limbs is just as simple as healing them

**Comment by Grazyn** [comment]
@xtdm the goal in combat is killing the other guy, not maiming him. I don't care I chopped off his limb if he still managed to win by outdamaging me.

\>use a sharp weapon
\>when their limb is chopped off they regain health and stop bleeding
 vs
\>use a blunt weapon
\>then their limb reaches threshold damage it's functionally chopped off BUT they don't regain health and don't stop bleeding

which one is better?

**Comment by XDTM** [comment]
If i make limb health not count towards actual health the heal on dismember is no longer a factor (and it was silly anyway).

Sharp weapons can dismember earlier with some rng, or they can disable them if they don't. Win-win.

If you care about long term damage you'll value dismemberment over bleeding, if you don't bleeding is not a factor.

Bleeding from dismemberment can be arranged in some way.

**Comment by Time-Green** [comment]
tactical amputation, don't fix it

**Comment by epochayur** [comment]
very cool, slowly but surely adding some depth to the medical system

i think @RandomMarine has the right idea

**Comment by XDTM** [comment]
Me too, i'll do it that way

**Comment by XDTM** [comment]
Suggestions applied! Limb damage is no longer "real" damage, dismemberment can happen before disabling making it a potential "crit" in combat, dismember-immune limbs (and species) don't suffer disabling either.

**Comment by Mickyan** [comment]
Played around with it a little bit, feels pretty good!

The main issue I've found is that it indirectly makes pressure, fire, heat and cold a lot less deadly since they only affect two body parts instead of six when it comes to going into crit.
Technically bombs too, although you're more likely to find yourself with more unusable limbs than usual instead of full crit.

And just personal preference but I'm not too keen on the sprites, how about a red outline instead of the red bits? (the red is too bright here but you get the idea)
screenshot_2 copy

**Comment by RandomMarine** [comment]
Yeah, damage from reagents and atmos and other things in that regard will need to be accounted for.
Also does significant (but not at full disability) leg/limb damage still cause slowdown?

**Comment by Shadowflame909** [comment]
I NEED IT IN SIMPLER TERMS FOR MY SMALL BRAIN. So if you get hurt at a certain point on your limbs now, you stop taking damage and you just cant use them?

**Comment by Fox-McCloud** [comment]
Don't think the "limbs don't contribute to full damage" is a good idea.

It dramatically alters the existing game in a way that I don't think will be positive. It means some simple mobs won't be able to ever effectively kill you; chemicals and other full-body damage things will be dramatically nerfed in terms of damage, as now they will only impact, health wise, two body-parts (ie: a chem that dealt 4 brute before, will now only deal a paltry 1.3 actual damage).

This will result in incredibly tanky humans, and only further entrench the "hit only head or body to kill", while making limbs more of a consolation prize (and lets be honest, how often are people going to *purely* want to just disable someone...not often).

Limbs failing when they're at full damage is an interesting concept; akin to Bay's broken bone system, though. 

The dismemberment changes are going to wind up being frustrating for players and admins both. Speaking from the experience of a codebase that *used to* operate like what you're implementing here (dismemberment can occur at any damage point other than 0). It tends to wind up with really stupid scenarios where someone has something like a pair of wirecutters and they whack someone in the head and kill them instantly, despite them having only 1-2 damage. It's absolutely hilarious the first few times you see it, but then it stops being funny and just starts being annoying. That's not only unbalanced, but will bring up a lot of grief cases of "so and so disembowled/beheaded/killed me just for shoving him once!" Again, I'm not speaking from a hypothetical here, I've seen this style dismemberment system play out.

**Comment by XDTM** [comment]
Good points overall, and short of reworking a lot of damage sources limbs should contribute to damage. A middle ground could be a multiplier, with limbs maybe being worth 75% of their damage.

Dismemberment from small amounts of damage will definitely be frustrating, so i'd implement a damage threshold at least (one hit delimbing from eswords is fine, and still not too frequent). Head dismemberment should still require the victim to be dead first, so that's not a problem.

**Comment by Razharas** [comment]
>var called "disabled"

When will we have a proper trauma system for fucks sake

**Comment by zamolxius** [comment]
@XDTM hey can you make it so grabbing a limb disables use of that limb? just a suggestion

**Comment by XDTM** [comment]
@Razharas one of my future plans is to make body traumas sorta like brain traumas

@zamolxius grabbing is currently not targeted, it'll take some refactoring to be doable... but two people holding someone by the arms while a third punches them in the stomach is something i want to see

**Comment by Ratvar** [comment]
Lasering/welding someone's legs for permastun becomes kiiiinda eeeaasy.

**Comment by XDTM** [comment]
Alright, about the whole limb damage thing:

IDEA: Limb damage does not count towards crit/slowdown
Why: 
- Makes sense, as limbs do not contain internal organs. The danger comes from bleeding, which is still a thing.
- Solves the realism issue of healing through dismemberment.
- Creates strategic gameplay of aiming to disable or aiming to crit.

Issues:
- Chemical Damage picks a random limb/all limbs: easy enough, it's internal damage and can be handwaved as being in the chest instead.
- Widespread Damage (from atmos, fire or explosions): all in all not a huge problem as long as we buff the damage in return. The issue is that it will disable all limbs at once, while it doesn't make too much sense to do so; might be solved by a percentage damage distribution with some randomization. (I.E. 60% to the chest, 10-15% to each limb, and so on)
- Animals targeting limbs: Probably solved by buffing either animal damage or bleeding. Bleeding in general is going to need a little overhaul as well, to make it actually dangerous.

From what i see it's doable and might push us a little closer to a DF-style vital-based health system, which i think might fit the game better than straight health.

If y'all approve of my solutions for the arising issues i'll get to coding and try to make it as balanced as possible.

Alternatively, i'll stick to the damage multiplier, which is much closer to the status quo and doesn't involve huge changes.

**Comment by CPTANT** [comment]
Love the disabling of limbs but limb damage contributing absolutely nothing to health damage isn't a good idea IMO.

I find the "realism" of limb damage not causing you to go crit rather dubious as passing out from pain is definitely a real thing and RL limb injuries cause all kinds of systemic effects.

The compromise of 75% or so is fine.

**Comment by XDTM** [comment]
Pain is generally not a factor in our other mechanics, though. I'd rather simulate pain separately, than half-mix it with physical damage.

**Comment by Ratvar** [comment]
What if mangled limbs deal part of their damage as stamina damage?

**Comment by XDTM** [comment]
@CPTANT, My roadmap for a potential full damage refactor is this:
- This limb refactor
- Bleeding rework, maybe usind wound components instead of just checking brute damage thresholds
- Some kind of pain system, which would cause limb injuries to hurt and have more consequences.
- A body trauma system (somewhat like brain traumas) to handle the systemic damage you mentioned, as well as more complex wounds and injuries in general.

What would you think?

**Comment by Mickyan** [comment]
While I like the idea of limb damage not counting for crit I'm starting to worry that there's so many things right now that rely on the health system being like it is, for every thing you account for there's likely ten others you're not considering

I think it would make more sense to go with damage multipliers for limbs right now and then come back to it along with whatever you have planned for the pain system

**Comment by XDTM** [comment]
Fair enough, it can indeed be done separately when everything is ready.

File: code/modules/surgery/bodyparts/bodyparts.dm [hunk 1]
```diff
@@ -124,10 +131,9 @@
 	brute = max(brute * dmg_mlt, 0)
 	burn = max(burn * dmg_mlt, 0)
 	stamina = max(stamina * dmg_mlt, 0)
-	if(status == BODYPART_ROBOTIC) //This makes robolimbs not damageable by chems and makes it stronger
-		brute = max(0, brute - 5)
-		burn = max(0, burn - 4)
-		//No stamina scaling.. for now..
+	brute = max(0, brute - brute_reduction)
+	burn = max(0, burn - burn_reduction)
+	//No stamina scaling.. for now..
```
Comment (Nabski89) [inline code/modules/surgery/bodyparts/bodyparts.dm]: This comment about the stamina scaling no longer make sense with robotic damage reduction being moved to the limbs.

Comment (XDTM) [reply code/modules/surgery/bodyparts/bodyparts.dm]: Oh yeah you're right

**Comment by optimumtact** [comment]
The fix for the damage caps on limbs is that when the limb is capped out the zone should be removed from valid targets and hits there target the chest instead

That fixes the concerns that fox has I believe

**Comment by Cyberboss** [comment]
Gonna throw this up for a while to see how it goes

**Comment by kevinz000** [comment]
i'm still burned out and dead so if the answer's in files changed and i missed it i'm sorry but @XDTM i remember doing something to make stamina damage limb specific, how will this factor in? does this mean security will have to disable someone's head and torso to actually get the stun, and/or does this mean they can disable someone's arms to make them unable to use objects?

**Comment by XDTM** [comment]
Stamina damage will count towards disabling, and when it goes back down to half-hp it will restore the functionality. If it all works properly you should be able to disable people's legs with disablers, and so on.

Oranges' suggestion is a good alternative, although i'm a little bit worried that it might remove all downsides from targeting limbs.

**Comment by XDTM** [comment]
That works, i'll try that

**Comment by vuonojenmustaturska** [comment]
it's been testmerged for a while and seems to be working

**Comment by XDTM** [comment]
Any more dismemberment oddities? I disabled beheading on living people (which apparently has always been enabled?) and lowered the base chance, i hope i haven't overdone it on the other direction.

**Comment by optimumtact** [comment]
I had to manually var edit their bodyparts to not be broken

**Comment by Supermichael777** [comment]
The 66% damage nerf to gun is rather negative

**Comment by XDTM** [comment]
Fixed augmentation, now it replaces the limb instead of just copying the robolimb's vars.

By the way, to fix that you just had to hit the limb again to trigger the health check

**Comment by adamkad1** [comment]
Head augumentation now kills the person because their head, and by consequence, their brain does not carry over

**Comment by XDTM** [comment]
Oops i forgot about that

**Comment by Cyberboss** [comment]
Fix conflicts ree

**Comment by optimumtact** [comment]
Break a  leg bud
